api/api/experiments.py

```diff
@@ -307,6 +315,27 @@ def create_experiment(experiment: Experiment):
         return exp
 
 
+@router.get("/{experiment_id}/stop")
+def stop_experiment(experiment_id: int):
+    with db.session() as session:
+        exp = session.get(Experiment, experiment_id)
+        if not exp:
+            raise HTTPException(status_code=404, detail="Experiment not found")
+        for task_id in exp.task_ids.split(","):
+            RUNNER.control.revoke(task_id, terminate=True)
+
+
+@router.delete("/{experiment_id}")
+def delete_experiment(experiment_id: int):
```

Comment:
Do the tasks need to be stopped if the experiment is deleted?

Reply:
yes! good catch